test/edu/stanford/dlss/wowza/TestParsingFromRequestInfo.java

```diff
@@ -0,0 +1,130 @@
+package edu.stanford.dlss.wowza;
+
+import static org.junit.Assert.*;
+
+import org.junit.*;
+
+
+/** test parsing of data from streamName and/or queryString */
+public class TestParsingFromRequestInfo
+{
+    SulWowza testModule;
+    static String stacksToken = "encryptedStacksMediaToken";
+    static String queryStr = "stacks_token=" + stacksToken;
+    static String streamName = "aa/000/aa/0000/mp4:example.mp4";
+    static String streamNameWithQueryStr = streamName + "?" + queryStr;
+
+    @Before
+    public void setUp()
+    {
+        testModule = new SulWowza();
+    }
+
+    // stacks token
+
+    // druid
+
+    // streamName
+
+    @Test
+    public void getStacksToken_queryStrOnly()
+    {
+        assertEquals(stacksToken, testModule.getStacksToken(queryStr));
+    }
+
+    @Test
+    public void getStacksToken_streamNameWithQueryStr()
+    {
+        assertEquals(stacksToken, testModule.getStacksToken(streamNameWithQueryStr));
+    }
+
+    @Test
+    public void getStacksToken_ignoresOtherQueryParams()
+    {
+        String myQueryStr = "ignored=ignored&stacks_token=" + stacksToken + "&anything=anythingElse";
+        assertEquals(stacksToken, testModule.getStacksToken(myQueryStr));
+        assertEquals(stacksToken, testModule.getStacksToken(streamName + "?" + myQueryStr));
+    }
+
+    @Test
+    public void getStacksTokenEmptyWhenEmptyTokenParam()
+    {
+        String myQueryStr = "ignored=ignored&stacks_token=&anything=anythingElse";
+        assertEquals("", testModule.getStacksToken(myQueryStr));
+        assertEquals("", testModule.getStacksToken(streamName + "?" + myQueryStr));
+    }
+
+    @Test
+    public void getStacksTokenNullWhenMissingTokenParam()
+    {
+        String myQueryStr = "ignored=ignored&anything=anythingElse";
+        assertNull(testModule.getStacksToken(myQueryStr));
+        assertNull(testModule.getStacksToken(streamName + "?" + myQueryStr));
+    }
+
+    @Test
+    public void getStacksTokenNullWhenEmptyStr()
+    {
+        assertNull(testModule.getStacksToken(""));
+    }
+
+    @Test
+    public void getStacksTokenNullWhenNullArg()
+    {
+        SulWowza testModule = new SulWowza();
+        assertNull(testModule.getStacksToken(null));
+    }
+
+    //@Test
+    public void getStacksToken_weirdChars()
+    {
+        // stacks provides a token that theoretically could have chars needing url encoding
+        //  does our code get those chars properly?
+        //  does it properly send them back to stacks/verify_token ?
+        fail("what if stacks tokens have chars that are utf-8 or need to be url encoded?");
```

Comment:
druid doesn't need to be tested because we know our druids won't need encoding.  Similarly for IP.